src/main/java/com/nike/cerberus/operation/core/RestoreCompleteCerberusDataFromS3BackupOperation.java

```diff
@@ -0,0 +1,344 @@
+            throw new RuntimeException("Failed to validate that the user wanted to proceed with backup", e);
+        }
+
+        if (! proceed.equalsIgnoreCase("proceed")) {
+            throw new RuntimeException("User did not confirm to proceed with backup restore");
+        }
+    }
+
+    private S3StoreService getS3EncryptionStoreService(String cmkId,
+                                                       RestoreCompleteCerberusDataFromS3BackupCommand command) {
+
+        Region region = Region.getRegion(Regions.fromName(command.getS3Region()));
+        KMSEncryptionMaterialsProvider materialProvider = new KMSEncryptionMaterialsProvider(cmkId);
+        AmazonS3EncryptionClient encryptionClient =
+                new AmazonS3EncryptionClient(
+                        new DefaultAWSCredentialsProviderChain(),
+                        materialProvider,
+                        new CryptoConfiguration()
+                                .withAwsKmsRegion(region))
+                        .withRegion(region);
+
+        return new S3StoreService(encryptionClient, command.getS3Bucket(), command.getS3Prefix());
+    }
+
+    private String getKmsCmkId(String path, S3StoreService s3StoreService) {
+        Map<String, String> metadata = s3StoreService.getS3ObjectUserMetaData(path);
+        if (! metadata.containsKey("x-amz-matdesc")) {
+            throw new RuntimeException("Failed to get Customer Master Key ID from object user metadata. " +
+                    "'x-amz-matdesc' not found in metadata for object at path: " + path);
+        }
+
+        String serializedEncryptionContext = metadata.get("x-amz-matdesc");
+
+        Map<String, String> encryptionContextMap;
+        try {
+            encryptionContextMap = objectMapper.readValue(serializedEncryptionContext,
+                    new TypeReference<HashMap<String,String>>() {});
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to convert encryption contect metadata value into Map");
+        }
+        return encryptionContextMap.get("kms_cmk_id");
+    }
+
+    private String getDecryptedJson(String sdbBackupKey, S3StoreService s3StoreService) {
+        Optional<String> json = s3StoreService.get(sdbBackupKey);
+        if (!json.isPresent()) {
+            logger.error("Failed to get json from S3 for {}", sdbBackupKey);
+        }
+        return json.get();
+    }
+
+    /**
+     * Process the stored backup json
+     * Step 1: Restores the metadata to CMS
+     * Step 2: Restores the secrete data to Vault
+     * @param sdbBackupJson the json string from s3
+     */
+    private void processBackup(String sdbBackupJson, CerberusAdminClient cerberusAdminClient) throws IOException {
+
+        JsonNode sdb = objectMapper.readTree(sdbBackupJson);
+
+        // restore metadata to cms
+        cerberusAdminClient.restoreMetadata(sdbBackupJson);
+        deleteAllSecrets(sdb.get("path").asText(), cerberusAdminClient);
```

Comment:
So this will delete the existing secrets, when doing a restore or else you would get a non-recursive left merge.
I also wonder if we should diff SDBs and delete the SDBs that exist in the destination but not in the backup?

Right now we get a weird merging situation where its not a total restore of a backup, but does it matter?